rfc/terraform-provider.md

```diff
@@ -0,0 +1,97 @@
+# Constellation Terraform Provider
+
+The Constellation Terraform Provider allows its user to manage the full lifecycle of a Constellation cluster -- namely creation, updates, and deletion -- via Terraform.
+
+## Design Goals
+
+- The User needs to be able to perform Creation, Upgrades and Deletion of a Cluster through the provider.
+- Infrastructure provisioning should explicitly **not** be performed by the Constellation Terraform provider.
+- The User needs to receive some sort of Access Token to the Cluster (e.g. a Kubeconfig, whether long- or short-lived)
+when applying a configuration containing the provider and the resources listed below.
+
```

Comment:
How would we include `status` information such as image / k8s upgrades? The user could manually fetch these with the CLI for troubleshooting cases? Same for `recovery`?

Reply:
The goal of the provider is to provide management and provisioning (i.e. create, update, delete) capabilities imo, not interaction with the cluster itself, like the CLI does. I don't think we need to achieve feature-parity with the CLI. The Terraform model would also not make it very easy to implement such without some hacks.

Reply:
I agree